yarn-project/aztec.js/src/utils/anvil_test_watcher.ts

```diff
@@ -60,23 +60,36 @@ export class AnvilTestWatcher {
     await this.filledRunningPromise?.stop();
   }
 
-  async mineIfSlotFilled() {
+  async warpTimeIfNeeded() {
     try {
       const currentSlot = await this.rollup.read.getCurrentSlot();
       const pendingBlockNumber = BigInt(await this.rollup.read.getPendingBlockNumber());
       const blockLog = await this.rollup.read.getBlock([pendingBlockNumber]);
+      const nextSlotTimestamp = Number(await this.rollup.read.getTimestampForSlot([currentSlot + 1n]));
 
       if (currentSlot === blockLog.slotNumber) {
         // We should jump to the next slot
-        const timestamp = await this.rollup.read.getTimestampForSlot([currentSlot + 1n]);
         try {
-          await this.cheatcodes.warp(Number(timestamp));
-          this.dateProvider?.setTime(Number(timestamp) * 1000);
+          await this.cheatcodes.warp(nextSlotTimestamp);
+          this.dateProvider?.setTime(nextSlotTimestamp * 1000);
         } catch (e) {
-          this.logger.error(`Failed to warp to timestamp ${timestamp}: ${e}`);
+          this.logger.error(`Failed to warp to timestamp ${nextSlotTimestamp}: ${e}`);
         }
 
         this.logger.info(`Slot ${currentSlot} was filled, jumped to next slot`);
+        return;
+      }
+
+      const currentTimestamp = this.dateProvider?.now() ?? Date.now();
+      if (currentTimestamp > nextSlotTimestamp * 1000) {
```

Comment:
Now that I think of it, shouldn't this be the `nextL1SlotTimestamp` instead? After all, what we want is that L1 keeps progressing, so we should mining L1 blocks at the expected L1 slot interval, not L2.